Neos.Utility.ObjectHandling/Classes/TYPO3/Flow/Reflection/ObjectAccess.php

```diff
@@ -97,13 +97,10 @@ protected static function getPropertyInternal($subject, $propertyName, $forceDir
     {
         if ($subject === null) {
             return null;
-        }
-        if (is_array($subject)) {
-            if (array_key_exists($propertyName, $subject)) {
-                $propertyExists = true;
-                return $subject[$propertyName];
-            }
-            return null;
+        } elseif (is_array($subject)) {
+            $propertySet = isset($subject[$propertyName]);
+            $propertyExists = $propertySet || array_key_exists($propertyName, $subject);
```

Comment:
Now I see what you mean. There are two options.
Option 1 (the one I used):
```php
$propertySet = isset($subject[$propertyName]);
$propertyExists = $propertySet || array_key_exists($propertyName, $subject);
return $propertySet ? $subject[$propertyName] : null;
```
Option 2 (the one you'd prefer):
```php
$propertyExists = isset($subject[$propertyName]) || array_key_exists($propertyName, $subject);
return $propertyExists ? $subject[$propertyName] : null;
```
It really does have the same effect since `$propertyExists && !$propertySet => $subject[$propertyName] === null`.